stan/math/prim/fun/multiply.hpp

```diff
@@ -93,22 +117,22 @@ inline Eigen::Matrix<return_type_t<T1, T2>, R1, C2> multiply(
  * specified column vector.  The return is the same as the dot
  * product.  The two vectors must be the same size.
  *
- * @tparam C1 number of columns in row vector, can be Eigen::Dynamic
- * @tparam R2 number of rows in column vector, can be Eigen::Dynamic
- * @tparam T1 type of elements in row vector
- * @tparam T2 type of elements in column vector
+ * @tparam T1 type of the row vector
+ * @tparam T2 type of the column vector
  *
  * @param rv row vector
  * @param v column vector
  * @return scalar result of multiplying row vector by column vector
  * @throw <code>std::invalid_argument</code> if rv and v are not the same size
  */
-template <int C1, int R2, typename T1, typename T2,
-          typename = require_all_arithmetic_t<T1, T2>>
-inline return_type_t<T1, T2> multiply(const Eigen::Matrix<T1, 1, C1>& rv,
-                                      const Eigen::Matrix<T2, R2, 1>& v) {
+template <
+    typename T1, typename T2,
+    typename = require_all_not_var_t<scalar_type_t<T1>, scalar_type_t<T2>>,
+    typename
+    = require_t<conjunction<is_eigen_row_vector<T1>, is_eigen_col_vector<T2>>>>
+inline auto multiply(const T1& rv, const T2& v) {
   check_matching_sizes("multiply", "rv", rv, "v", v);
-  return rv.dot(v);
+  return dot_product(rv, v);
```

Comment:
This change allows using same overload for prim and fwd.

Reply:
I wonder if the dot product specialization for multiply was because previously we were only allowing Eigen dynamic matrices in some places and internally eigen would just call dot product if it saw a row and column vector

Reply:
I am pretty sure the reason was to return a scalar instead of 1x1 matrix.